modules/data.remote/R/NASA_DAAC_download.R

```diff
@@ -298,30 +305,34 @@ NASA_CMR_finder <- function(doi) {
   return(as.list(data.frame(cbind(provider, concept_id))))
 }
 
-#' Set NASA DAAC credentials to the current environment.
+#' Set NASA DAAC credentials to the .netrc file.
 #'
-#' @param replace Boolean: determine if we want to replace the current credentials from the environment. The default is FALSE.
-#' @param folder.path Character: physical path to the folder that contains the credential file. The default is NULL.
+#' @param dl_dir Character: physical path to the folder that the .netrc file will be generated.
 #'
 #' @author Dongchen Zhang
-DAAC_Set_Credential <- function(replace = FALSE, folder.path = NULL) {
-  if (replace) {
-    PEcAn.logger::logger.info("Replace previous stored NASA DAAC credentials.")
-  }
-  # if we have the credential file.
-  if (!is.null(folder.path)) {
-    if (file.exists(file.path(folder.path, ".nasadaacapirc"))) {
-      key <- readLines(file.path(folder.path, ".nasadaacapirc"))
-      Sys.setenv(ed_un = key[1], ed_pw = key[2])
-    }
-  }
-  # otherwise we will type the credentials manually.
-  if (replace | nchar(Sys.getenv("ed_un")) == 0 | nchar(Sys.getenv("ed_un")) == 0) {
-    Sys.setenv(ed_un = sprintf(
-      getPass::getPass(msg = "Enter NASA Earthdata Login Username \n (or create an account at urs.earthdata.nasa.gov) :")
-    ), 
-    ed_pw = sprintf(
-      getPass::getPass(msg = "Enter NASA Earthdata Login Password:")
-    ))
+getnetrc <- function (dl_dir) {
+  netrc <- file.path(dl_dir, ".netrc")
+  netrc <- gsub("~", Sys.getenv("HOME"), netrc)
```

Comment:
1. I suggest allowing arbitrary filenames in case users want separate netrcs per host/profile (I guess this means renaming `credential_folder` to `credential_path` above too)
2. technically only a leading tilde should expand to HOME; path.expand knows the details so we don't have to.
```suggestion
getnetrc <- function (dl_path) {
  netrc <- path.expand(dl_path)
```

Reply:
Fixed.